### ISSUE TITLE
Broken output of containers

### ISSUE THREAD

**Issue by YarikTH** [sharkdp/dbg-macro#102]
See example below.
https://godbolt.org/z/vnxaqq
```cpp
#include <dbg.h>

int main()
{
    std::vector<std::pair<std::string, int>> container = {
        {"30+2", 32}, {"30-2", 28}, {"30*2", 60}
    };
    dbg(container);
}
```
Here we have a vector of pairs of strings. Despite the library has pretty_print overloads for std::pair, std::string, and container they don't fully work with each other. It seems that the reason is the fact, that pretty_print overloads for std::pair and std::string are defined just after container overload and thus they aren't visible from pretty_print overload for containers.

A possible solution is adding declarations for all the pretty_print overloads before their definitions. For e.g. in my example, adding declarations for pair and string overloads before container overload fixes the problem.

```cpp
inline bool pretty_print(std::ostream& stream, const std::string& value);

template <typename T1, typename T2>
inline bool pretty_print(std::ostream& stream, const std::pair<T1, T2>& value);



template <typename Container>
inline typename std::enable_if<detail::is_container<const Container&>::value,
                               bool>::type
pretty_print(std::ostream& stream, const Container& value) {
```

**Comment by sharkdp** [comment]
Thank you for reporting this. That should be fixed.

If there is a solution without duplicating all declarations, that would be great. But I guess that's actually the right thing to do.

**Comment by YarikTH** [comment]
I think that alternative is using one big function pretty_print with constexpr_if branches. I used it in another project. Easy to maintain, easy to understand. But need to bump std to c++17. I think that for this project it's unwanted.

**Comment by sharkdp** [comment]
Yes. I would like to maintain C++11 backwards compatibility.

**Comment by sharkdp** [comment]
> Hi, are you planning to fix that issue?

Yes, this should be fixed. We can go the easy way for now and duplicate all declarations. Any help would be very much appreciated.
